qcow2reader_test.go

```diff
@@ -0,0 +1,155 @@
+package qcow2reader
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+type CompressionType string
```

Comment:
nit: use https://github.com/lima-vm/go-qcow2reader/blob/63138be42eade73a48a657be80e43e2119847a5f/image/qcow2/qcow2.go#L168

Reply:
This is int 0 - but for qemu-img we need a string, so I don't think it helps, unless CompressionType will have a String() method returning "zlib" for this constant.

Reply:
Probably you can have `map[qcow2.CompressionType]string` in `qcow2reader_test.go`

Reply:
Or maybe just use a raw string without defining `type CompressionType string`?

It is confusing to have two different definitions of `CompressionType`

Reply:
Trying...